lua/cmp/utils/window.lua

```diff
@@ -104,45 +126,35 @@ window.update = function(self)
     local info = self:info()
     local bar_height = math.ceil(info.height * (info.height / total))
     local bar_offset = math.min(info.height - bar_height, math.floor(info.height * (vim.fn.getwininfo(self.win)[1].topline / total)))
-    local style1 = {}
-    style1.relative = 'editor'
-    style1.style = 'minimal'
-    style1.width = 1
-    style1.height = info.height
-    style1.row = info.row
-    style1.col = info.col + info.width - (info.has_scrollbar and 1 or 0)
-    style1.zindex = (self.style.zindex and (self.style.zindex + 1) or 1)
-    if self.swin1 and vim.api.nvim_win_is_valid(self.swin1) then
-      vim.api.nvim_win_set_config(self.swin1, style1)
-    else
-      style1.noautocmd = true
-      self.swin1 = vim.api.nvim_open_win(buffer.ensure(self.name .. 'sbuf1'), false, style1)
-      vim.api.nvim_win_set_option(self.swin1, 'winhighlight', 'EndOfBuffer:PmenuSbar,Normal:PmenuSbar,NormalNC:PmenuSbar,NormalFloat:PmenuSbar')
-    end
+    local border_offset_row, border_offset_col = window.border_offset_scrollbar(self.style)
+
     local style2 = {}
     style2.relative = 'editor'
     style2.style = 'minimal'
     style2.width = 1
     style2.height = bar_height
-    style2.row = info.row + bar_offset
-    style2.col = info.col + info.width - (info.has_scrollbar and 1 or 0)
+    style2.row = info.row + bar_offset + border_offset_row
+    style2.col = info.col + info.width - (info.has_scrollbar and 1 or 0) - border_offset_col
     style2.zindex = (self.style.zindex and (self.style.zindex + 2) or 2)
-    if self.swin2 and vim.api.nvim_win_is_valid(self.swin2) then
-      vim.api.nvim_win_set_config(self.swin2, style2)
+    if self.swin and vim.api.nvim_win_is_valid(self.swin) then
+      vim.api.nvim_win_set_config(self.swin, style2)
     else
       style2.noautocmd = true
-      self.swin2 = vim.api.nvim_open_win(buffer.ensure(self.name .. 'sbuf2'), false, style2)
-      vim.api.nvim_win_set_option(self.swin2, 'winhighlight', 'EndOfBuffer:PmenuThumb,Normal:PmenuThumb,NormalNC:PmenuThumb,NormalFloat:PmenuThumb')
-    end
-  else
-    if self.swin1 and vim.api.nvim_win_is_valid(self.swin1) then
-      vim.api.nvim_win_hide(self.swin1)
-      self.swin1 = nil
-    end
-    if self.swin2 and vim.api.nvim_win_is_valid(self.swin2) then
-      vim.api.nvim_win_hide(self.swin2)
-      self.swin2 = nil
+      local sbuf2 = buffer.ensure(self.name .. 'sbuf2')
+      self.swin = vim.api.nvim_open_win(sbuf2, false, style2)
+      local highlight = self.scrollbar ~= '' and 'CmpItemMenuThumb' or 'PmenuThumb'
+      vim.api.nvim_win_set_option(self.swin, 'winhighlight', 'EndOfBuffer:'..highlight..',NormalFloat:'..highlight)
+
+      if self.scrollbar ~= '' then
+        local replace = {}
+        for i = 1, style2.height do replace[i] = '║' end
```

Comment:
for i = 1, style2.height do replace[i] = self.scrollbar end

Reply:
wouldn't you need to pass scrollbar here?

Reply:
Oops. I hit undo on the keyboard before committing. Good catch!